src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AzureFunction/AzureFunctionInProcessInvokeAsyncWrapper.cs

```diff
@@ -0,0 +1,252 @@
+        {
+            try
+            {
+                if (responseTask.IsFaulted)
+                {
+                    transaction.NoticeError(responseTask.Exception);
+                    return;
+                }
+
+                var result = GetTaskResult(responseTask);
+                if (result == null)
+                {
+                    return;
+                }
+
+                var resultType = result.GetType();
+                agent.Logger.Debug($"Azure Function response type: {resultType.FullName}");
+
+                // insert DT headers if the response object is a ServiceBusMessage
+                if (resultType.FullName == "Azure.Messaging.ServiceBus.ServiceBusMessage")
+                {
+                    TryInsertServiceBusDTHeaders(transaction, result);
+                    return;
+                }
+
+                // if the trigger is HTTP, try to set the StatusCode
+                if (trigger == "http" && handledHttpArg)
+                {
+                    TrySetHttpResponseStatusCode(result, resultType, transaction, agent);
+                }
+            }
+            finally
+            {
+                segment.End();
+            }
+        }
+    }
+
+    private bool TrySetHttpResponseStatusCode(dynamic result, Type resultType, ITransaction transaction, IAgent agent)
+    {
+        // make sure there's a StatusCode property on the result object
+        var statusCodeProperty = resultType.GetProperty("StatusCode");
+        if (statusCodeProperty != null)
+        {
+            var statusCode = statusCodeProperty.GetValue(result);
+            if (statusCode != null)
+            {
+                transaction.SetHttpResponseStatusCode((int)statusCode);
+                return true;
+            }
+        }
+
+        agent.Logger.Debug($"Could not find StatusCode property on response object type {result?.GetType().FullName}");
+        return false;
+    }
+
+    private bool TryHandleHttpTrigger(dynamic arg, string argTypeName, ITransaction transaction, IAgent agent)
+    {
+        if (arg is System.Net.Http.HttpRequestMessage httpTriggerArg)
+        {
+            IEnumerable<KeyValuePair<string, IEnumerable<string>>> headers = httpTriggerArg.Headers; // IEnumerable<KeyValuePair<string, IEnumerable<string>>>
+
+            Uri requestUri = httpTriggerArg.RequestUri;
+            string requestPath = requestUri.AbsolutePath;
+            transaction.SetUri(requestPath);
+
+            string requestMethod = httpTriggerArg.Method.Method;
+            transaction.SetRequestMethod(requestMethod);
+
+            if (headers != null)
+            {
+                transaction.AcceptDistributedTraceHeaders(headers, GetHeaderValueFromIEnumerable, TransportType.HTTP);
+            }
+
+            return true;
+        }
+
+        if (argTypeName is "Microsoft.AspNetCore.Http.DefaultHttpRequest" or "Microsoft.AspNetCore.Http.HttpRequest")
```

Comment:
In some places this comparison is done using `==` and here it is done using `is`. We should be more consistent in how we do this comparison.

Reply:
`is` is the same thing as `=='; the `is` bit is just syntactic sugar. I'm fine with using whichever makes sense for the line it's on.